jbehave-support-core/src/main/java/org/jbehavesupport/core/report/XmlReporterFactory.java

```diff
@@ -157,9 +160,47 @@ private void copyResourceToReportDirectory(Resource resource) {
     private void copyResourceToReportDirectory(Resource resource, String targetFilePath) {
         File targetFile = new File(targetFilePath);
         try {
-            FileUtils.copyInputStreamToFile(resource.getInputStream(), targetFile);
+            String resourcePath = String.valueOf(resource.getURI());
+            String jarPath = resourcePath.replace(("!/" + resource.getFilename()), "").replace("jar:file:", "");
+            File jarFile = new File(jarPath);
+
+            if (jarFile.isFile()) {
+                copyResourceFromJar(resource, targetFilePath, targetFile, jarFile);
+            } else {
+                copyResourceFromFile(resource, targetFile);
+            }
         } catch (IOException e) {
-            log.error("Cannot copy resource file to target", e);
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void copyResourceFromFile(Resource resource, File targetFile) throws IOException {
+        if (resource.isFile() && resource.getFile().isDirectory()) {
+            FileUtils.copyDirectory(resource.getFile(), targetFile);
+        } else {
+            FileUtils.copyInputStreamToFile(resource.getInputStream(), targetFile);
+        }
+    }
+
+    private void copyResourceFromJar(Resource resource, String targetFilePath, File targetFile, File jarFile) throws IOException {
```

Comment:
`targetFilePath` and `targetFile` are effectively the same, can be reduced to one.